libs/ui-toolkit/src/components/transaction-dialog/transaction-dialog.stories.tsx

```diff
@@ -0,0 +1,62 @@
+import type { ComponentStory, ComponentMeta } from '@storybook/react';
+import { TransactionDialog } from './transaction-dialog';
+import { TxState } from '@vegaprotocol/react-helpers';
+
+export default {
+  component: TransactionDialog,
+  title: 'TransactionDialog',
+  parameters: {
+    themes: false,
+  },
+} as ComponentMeta<typeof TransactionDialog>;
+
+const Template: ComponentStory<typeof TransactionDialog> = (args) => (
+  <TransactionDialog {...args} />
+);
+
+export const Requested = Template.bind({});
+Requested.args = {
+  name: 'Some tx',
+  status: TxState.Requested,
+  error: null,
+  confirmations: 0,
+  txHash: null,
+  requiredConfirmations: 1,
+  confirmed: false,
+};
+
+export const Pending = Template.bind({});
+Pending.args = {
+  name: 'Some tx',
+  status: TxState.Pending,
+  error: null,
+  confirmations: 1,
+  txHash: '0x123123',
+  requiredConfirmations: 3,
+  confirmed: false,
+};
+export const Error = Template.bind({});
+Error.args = {
+  name: 'Some tx',
+  status: TxState.Error,
+  error: {
+    name: 'Error',
+    message:
+      'cannot estimate gas; transaction may fail or may require manual gas limit [See: https://links.ethers.org/v5-errors] {"code": 3,"data":"0x08c379000000000000000000000000000000000000000000000000000000000" }',
```

Comment:
question (non-blocking): can we test this?

Reply:
Not sure of a good way to unit test this tbh or even e2e test without some sort of visual regression tool?